src/main/java/com/sleekbyte/tailor/listeners/CommentAnalyzer.java

```diff
@@ -33,6 +30,8 @@ public CommentAnalyzer(CommonTokenStream tokenStream, Printer printer) {
         extractComments();
     }
 
+    public abstract void analyze();
+
     private void extractComments() {
         for (Token token : tokenStream.getTokens()) {
```

Comment:
I think this process of extracting comments is expensive, so maybe we want a class dedicated to doing just that? We could extract the comments once and then we could just pass the list of comment tokens around among the different comment related listeners?